.env.example

```diff
@@ -170,6 +170,10 @@ export BOOKWAREHOUSE_NAMESPACE=bookwarehouse
 # Default: 90s
 # export TIMEOUT=90s
 
+### optional: Whether the CTR_REGISTRY is a private container registry
+# Default: false
+#export USE_PRIVATE_REGISTRY=false
```

Comment:
@nojnhuh do you think this should be true or false by default? I think we should match whatever kind needs. Do we need the ctr_registry creds for kind?

Reply:
The demo itself doesn't manage kind clusters like the e2e tests do, the demo always assumes you have _some_ cluster up and running beforehand. I usually do `make kind-up` which also ensures a local image registry is running and makes it available to the kind cluster. That registry doesn't require credentials and since it's the default in .env.example I think false for this by default makes the most sense.

I admit I haven't been following this PR very closely, but scripts/create-container-registry-creds.sh already checks whether or not the image pull secret needs to be created based on if `$CTR_REGISTRY_PASSWORD` is defined: https://github.com/openservicemesh/osm/blob/de4c5b9eb941aa09147005be0224d6c686908022/scripts/create-container-registry-creds.sh#L16

Does `$USE_PRIVATE_REGISTRY` mean that same thing? If so, I don't have a preference which we use to check if the secrets are needed, but it would be worth making that consistent.

Reply:
We thought that using a variable like USE_PRIVATE_REGISTRY would be a little more explicit and intuitive than checking to see if the CTR_REGISTRY_PASSWORD is null. But yeah it essentially does the same thing. 

I also opened this PR separately to use the USE_PRIVATE_REGISTRY variable in other scripts as well: https://github.com/openservicemesh/osm/pull/3876

Reply:
could someone please explain the intention of this change ? If openshift specifically doesn't need a private registry then this can be explicitly be set to false for that particular scenario and not for the demo in general.

If someone is using a private registry for development purpose, which is likely the case ( excluding kind) this will cause issues.

Reply:
Further while making such changes please ensure that it is tested not only on kind but on other cluster types as well

Reply:
If scripts/create-container-registry-creds.sh already checks whether or not the image pull secret needs to be created based on if $CTR_REGISTRY_PASSWORD is defined, please define USE_PRIVATE_REGISTRY based on that (by adding custom logic) and do not define another default value, expecting users to set another variable in their .env.

Reply:
Discussed offline with Sneha, we are thinking it might be better to set the default value of USE_PRIVATE_REGISTRY to true.